source/main/windows/windowBounds.js

```diff
@@ -0,0 +1,110 @@
+// @flow
+import { debounce } from 'lodash-es/function';
+
+interface Rectangle {
+  x: number;
+  y: number;
+  width: number;
+  height: number;
+}
+
+interface Window {
+  getBounds(): Rectangle;
+  on(event: string, handler: Function): void;
+}
+
+interface Display {
+  workArea: Rectangle;
+}
+
+interface Screen {
+  getDisplayMatching(rect: Rectangle): Display;
+  workArea: Rectangle;
+}
+
+interface Store {
+  set: (key: string, value: Rectangle) => void;
+  get: (key: string) => ?Rectangle;
+}
+
+const windowBoundsDefaultStoreKey = 'windowBounds';
+
+function getRightEdge(rect: Rectangle) {
```

Comment:
Is there a reason why we don't use arrow functions?

Reply:
@danielmain yeah, I stopped using arrow functions if not necessary - the standard function declaration has several benefits like zero overhead (no context binding) and debugging - these functions have proper names in stack traces etc.

Reply:
IMHO: I thought arrow functions is feature and good practice to avoid context binding by making functions not pure. But of course the old fashion is still valid 😀

Reply:
using functional programming style you will never use context, so arrow functions are useless except for anonymous inline functions where the abbreviation is a real feature 😉 of course arrow functions are a language feature - but they also come with downsides (e.g: being anonymous / harder to debug by default)